### DISCUSSION TITLE
Description is empty if first element is not a paragraph

### DISCUSSION THREAD

**Discussion by jotona** [nuxt/content#1285] [Q&A]
The default description returned from queryContent is an empty string if the first element in the markdown file is not a paragraph (in this case it's a H2).

Is this expected behaviour?

In my use case, it's for a blog, and the markdown files also include a YAML Front Matter at the top.

**Comment by ManasMadrecha** [comment]
Isn't that logical? If you add a table or an image initially (or even a heading), how will the Content Module know what's the description? Whereas, having a `paragraph`'s text turn into `description` is more intuitive.

And, any way, even if your file's first element is heading, you still add any description in the frontmatter you want manually. If you still want automatic description to be added even if your first element is not a paragraph, but a heading, you can do so with a custom remark plugin.

E.g., `visit` the `paragraph` (or any `node` you prefer) whose `idx` is first, and grab its `textContent` using the apt `unist-util-**`. And assign this text to the `data.description` if the description is empty.

**Comment by nobkd** [comment]
You can try using the `<!--more-->` divider to define where the excerpt / description ends. I haven't used it, but you could try it out.
Here is a [PR for the docs](https://github.com/nuxt/content/pull/1441/files#diff-5c211214de8321c083373be581980078f8cce8b72833ee24565799eb79690f0dR75-R89)

EDIT: Here is the documentation entry, since the PR was merged: https://content.nuxtjs.org/guide/writing/markdown#excerpt
